app/proguard.cfg

```diff
@@ -102,6 +104,7 @@
 -dontnote okhttp3.**
 
 #Retrofit Stuff
+-keep class retrofit2.** { *; }

```

Comment:
How does Gradle update affect proguard rules ?

Reply:
Gradle 8 is on [full mode](https://r8.googlesource.com/r8/+/refs/heads/master/compatibility-faq.md#r8-full-mode) by default, which means more aggressive optimizations. In this case, it was causing `IllegalArgumentExceptions` with the message `Call return type must be parameterized as Call<Foo> or Call<? extends Foo>` during get requests, which made me suspect that maybe some references had been obfuscated. After some digging I found [some posts](https://stackoverflow.com/questions/64439204/call-return-type-must-be-parameterized-as-callfoo-or-call-extends-foo-excep) that pointed that way too.

Reply:
got it, although I don't think it's a good practice to retain all code from a package. Probably we should instead take the rules from [here](https://github.com/square/retrofit/blob/trunk/retrofit/src/main/resources/META-INF/proguard/retrofit2.pro)

Reply:
looks like this is still pending.

Reply:
yes, just pushed that also